src/ShallowTraversal.js

```diff
@@ -66,6 +69,18 @@ export function nodeHasId(node, id) {
   return propsOfNode(node).id === id;
 }
 
+
+export function nodeHasProperty(node, propKey, propValue) {
+  const nodeProps = propsOfNode(node);
+
+  if (propValue) {
+    return nodeProps[propKey] === propValue;
+  }
+
+  return nodeProps.hasOwnProperty(propKey);
```

Comment:
In fact, I just tested this with the new changes and it doesn't work in this situation.

```js
wrapper = <div foo={false} />

wrapper.find('[foo=false]')
```
In that situation that turns into 
```js
!!wrapper['foo'] // false
```
which then says it doesn't have the prop, though it does. So hasOwnProperty seems to be the way to go.